cloud-sql/mysql/Startup.cs

```diff
@@ -55,152 +54,157 @@ public void ConfigureServices(IServiceCollection services)
+                // Equivalent connection string:
+                // "Server=<dbSocketDir>/<INSTANCE_CONNECTION_NAME>;Uid=<DB_USER>;Pwd=<DB_PASS>;Database=<DB_NAME>;Protocol=unix"
+                String dbSocketDir = Environment.GetEnvironmentVariable("DB_SOCKET_PATH") ?? "/cloudsql";
+                String instanceConnectionName = Environment.GetEnvironmentVariable("INSTANCE_CONNECTION_NAME");
+                var connectionString = new MySqlConnectionStringBuilder()
                 {
-                    TimeSpan.FromSeconds(1),
-                    TimeSpan.FromSeconds(2),
-                    TimeSpan.FromSeconds(5)
-                }, (result, timeSpan, retryCount, context) =>
+                    // The Cloud SQL proxy provides encryption between the proxy and instance.
+                    SslMode = MySqlSslMode.None,
+                    // Remember - storing secrets in plaintext is potentially unsafe. Consider using
+                    // something like https://cloud.google.com/secret-manager/docs/overview to help keep
+                    // secrets secret.
+                    Server = String.Format("{0}/{1}", dbSocketDir, instanceConnectionName),
+                    UserID = Environment.GetEnvironmentVariable("DB_USER"),   // e.g. 'my-db-user
+                    Password = Environment.GetEnvironmentVariable("DB_PASS"), // e.g. 'my-db-password'
+                    Database = Environment.GetEnvironmentVariable("DB_NAME"), // e.g. 'my-database'
+                    ConnectionProtocol = MySqlConnectionProtocol.UnixSocket
+                };
+                connectionString.Pooling = true;
+                // Specify additional properties here.
+                // [START_EXCLUDE]
+                SetDbConfigOptions(connectionString);
+                // [END_EXCLUDE]
+                return connectionString;
+                // [END cloud_sql_mysql_dotnet_ado_connection_socket]
+            }
+
+            public MySqlConnectionStringBuilder GetMySqlConnectionString()
+            {
+                if (Environment.GetEnvironmentVariable("DB_HOST") != null)
                 {
-                    // Log any warnings here.
-                })
-                .Execute(() =>
+                    return NewMysqlTCPConnectionString();
+                }
+                else
                 {
-                    // Return a new connection.
-                    // [START_EXCLUDE]
-                    if (Environment.GetEnvironmentVariable("DB_HOST") != null)
-                    {
-                        return NewMysqlTCPConnection();
-                    }
-                    else
-                    {
-                        return NewMysqlUnixSocketConnection();
-                    }
-                    // [END_EXCLUDE]
-                });
-            // [END cloud_sql_mysql_dotnet_ado_backoff]
-            return connection;
+                    return NewMysqlUnixSocketConnectionString();
+                }

```

Comment:
(suggestions are not working for me here) but:

save the connection to be returned in a variable and then, before returning call `SetDbConfigOptions(connectionString);` that way, you can remove the call (and the "exclude" tags) from the other two methods.

Reply:
Done.